integrations/tensorflow/iree_tf_compiler/TFL/LowerGlobalTensors.cpp

```diff
@@ -112,85 +110,72 @@ class LowerGlobalTensorsPass
         loc = std::get<1>(*locIt);
       }
 
-      std::string flowSymName = "__iree_flow_" + name.str();
-
       // TODO(suderman): Determine the global type based on all store
       // operations.
-      auto global = builder.create<iree_compiler::IREE::Input::GlobalOp>(
-          loc, flowSymName, /*is_mutable=*/true, attribute.getType(),
-          attribute);
+      std::string symName = "__global_" + name.str();
+      auto global = builder.create<mlir::ml_program::GlobalOp>(
+          loc, symName, attribute.getType(), /*is_mutable=*/true, attribute,
+          nullptr);
       global.setPrivate();
-      sharedNameToFlowName[name] = std::move(flowSymName);
-    }
 
-    // Replace handles with global addresses.
-    for (auto handle : handleOps) {
-      auto name = handle.shared_name();
-      auto flowName = sharedNameToFlowName[name];
-      auto constIt = sharedNameToConstant.find(name);
-      if (constIt == sharedNameToConstant.end()) continue;
-
-      auto attribute = std::get<1>(*constIt);
-
-      builder.setInsertionPoint(handle);
-      auto address =
-          builder.create<iree_compiler::IREE::Input::GlobalAddressOp>(
-              handle.getLoc(),
-              iree_compiler::IREE::Input::PtrType::get(context,
-                                                       attribute.getType()),
-              SymbolRefAttr::get(builder.getContext(), flowName));
-      handle.getResult().replaceAllUsesWith(address.getResult());
-      handle.erase();
+      symbolRefMap[name] = global;
     }
 
     // Replace the assign ops with a global store operation.
     for (auto assign : assignOps) {
-      auto address = dyn_cast<iree_compiler::IREE::Input::GlobalAddressOp>(
+      auto handle = dyn_cast<mlir::TFL::VarHandleOp>(
```

Comment:
To my knowledge TFLite only AssignOp and ReadOp, which correspond to indirect read/write. In some cases we can convert it to the `ml_program.global_read` or `ml_program.global_write` when the above case is true.

Reply:
Sorry I misspoke, in which cases is the defining op not VarHandle? (I think control flow and func args are there and in those cases I'm guessing the pass fails)

Reply:
So overall VarHandle is the only thing that can create the resource variables for Assign and Read. The pass would fail on control flow / func but I *think* they should technically never happen because of TFLite's restrictive format. I can add some checks so the failure is more graceful but it may never occur in practice.

Reply:
As long as it fails in a loud and obvious manner if this isn't met (in this or next pass) I'm fine. I just don't want it to be easy to miss.